src/test/java/org/alfresco/service/cmr/repository/datatype/DefaultTypeConverterTest.java

```diff
@@ -118,8 +120,12 @@ public void testToString()
         // VersionNumber
         assertEquals("1.2.3", DefaultTypeConverter.INSTANCE.convert(String.class, new VersionNumber("1.2.3")));
         // Period
-        assertEquals("period", DefaultTypeConverter.INSTANCE.convert(String.class, new Period("period")));
+        //assertEquals("period", DefaultTypeConverter.INSTANCE.convert(String.class, new Period("period")));
         assertEquals("period|12", DefaultTypeConverter.INSTANCE.convert(String.class, new Period("period|12")));
+        Map<String,String> periodMap = new HashMap<String, String>();
+        periodMap.put("periodType","month");
+        periodMap.put("expression","1");
+        assertEquals("month|1", DefaultTypeConverter.INSTANCE.convert(String.class, new Period(periodMap)));
```

Comment:
This is testing Period to String conversion, which is already well tested. It is really testing the map constructor, which is sort of ok, but would be better in a PeriodTest class.

Reply:
I will add a test in PeriodTest you are right